hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java

```diff
@@ -296,7 +297,7 @@ private void checkTransactionDeprecation(@NonNull final Transaction tx) throws P
             // If they didn't use `signedTransactionBytes` and they didn't use `bodyBytes` then they didn't send a body
             // NOTE: If they sent a `sigMap` without a `bodyBytes`, then the `sigMap` will be ignored, just like
             // `body` and `sigs` are. This isn't really nice but not fatal.
-            throw new PreCheckException(INVALID_TRANSACTION_BODY);
+            throw new PreCheckException(NOT_SUPPORTED);
```

Comment:
This doesn't seem right. If there is no transaction body then it seems like INVALID_TRANSACTION_BODY is the correct response. NOT_SUPPORTED is supposed to be for when the transaction or query type is deprecated or never implemented.

Reply:
This is to follow mono service behavior. INVALID_TRANSACTION_BODY failed on one of the test in this suite.

Reply:
@iwsimon Did you see logic somewhere that threw `NOT_SUPPORTED` ? 
I see [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/submission/StructuralPrecheck.java#L88) it throws `INVALID_TRANSACTION_BODY`

Reply:
@Neeharika-Sompalli this test:
failsForMissingPayment() {
        return defaultHapiSpec("FailsForMissingPayment")
                .given()
                .when()
          .then(getAccountInfo(GENESIS).useEmptyTxnAsAnswerPayment().hasAnswerOnlyPrecheck(NOT_SUPPORTED));
    }
 expects NOT_SUPPORTED.

Reply:
Actually that test is failing in mono-service also. It should have `INVALID_TRANSACTION_BODY` 😄

Reply:
Set back to INVALID_TRANSACTION_BODY and updated the test with INVALID_TRANSACTION_BODY precheck.
Now we should run mono code to see if the test passed before we check module code😀

Reply:
True unfortunately these tests are not run in mono-service 😄